### PR TITLE
misc fixes

### LINKED ISSUE #2996: Fetchmail fails on folders with German umlauts
## Environment & Version

- Docker Compose version v2.20.0
- MailU version 2.0

## Description
When trying to fetch the folder `Entwürfe` from it results in an error. When you encode the name manually to `Entw&APw-rfe` according to the IMAP UTF-7 encoding scheme ([see here](https://stackoverflow.com/a/45787169/4239139)) it works.

## Replication Steps
- Add a fetchmail configuration, e.g. a web.de account
- Add the folder `Entwürfe` to be fetched
- Wait for the cronjob to execute

## Observed behaviour
- fetchmail runs into an ERROR

## Expected behaviour
- fetchmail should work just fine by encoding the folders to be fetched beforehand

## Logs
```
mailu-fetchmail-1  | 2023-10-22T13:26:12.475249522Z fetchmail: mailbox selection failed
mailu-fetchmail-1  | 2023-10-22T13:26:12.481254947Z fetchmail: client/server synchronization error while fetching from <private>@web.de@imap.web.de
mailu-fetchmail-1  | 2023-10-22T13:26:12.491866138Z Sleeping for 600 seconds
mailu-fetchmail-1  | 2023-10-22T13:26:12.491939658Z fetchmail: Query status=7 (ERROR)
```

### LINKED ISSUE #3272: Fastcgi header size in nginx webmail configuration
## Environment & Version

- `docker compose version` : Docker Compose version v2.27.0
- Version: `master` : 2.0

## Description
When trying to import contacts from a csv file using roundcube I get a 502 Bad Gateway message inside the import popup.
The file is not huge, barely 55k and ~ 400 contacts.

image

After a few digging in the stack I found this message in the webmail container logs : 

image

## Replication Steps
Launch a new stack and try and import contacts from a big enough csv file. Since the next step should be mapping the column names I'm unsure if the number of columns in the csv may be more relevant than the number of lines ...
For information this csv comes from a Yahoo Mail export and contains 34 columns.

## Workaround
I mounted a volume on the /conf directory from the container. Edited the nginx-webmail.conf file by adding the following lines in the server section : 
```
    # fastcgi buffers for php-fpm #
    fastcgi_buffers 16 32k;
    fastcgi_buffer_size 64k;
    fastcgi_busy_buffers_size 64k;

    # nginx buffers #
    proxy_buffer_size 128k;
    proxy_buffers 4 256k;
    proxy_busy_buffers_size 256k;
````
Restarted the container, and ... Voila !!!

### LINKED ISSUE #3379: Create user is not possible when domain quota and environment variable DEFAULT_QUOTA are set
## Environment & Version

- `k3s/kubernetes`
- Version: `2024.06.8`

## Description

In version 2024.06.8 it is not possible to create  a user in a domain with a maximum quota set, if the environment variable  DEFAULT_QUOTA is set in the admin container.

The reason is [line 27 in users.py](https://github.com/Mailu/Mailu/blob/d3cee8d3f26f2fcac06c8764e5da8cff8e789780/core/admin/mailu/ui/views/users.py#L27), because the variable DEFAULT_QUOTA is read from the environment as a string.

The problem occurs on with the helm chart as DEFAULT_QUOTA is set.

My temporary solution is to cast the variable to an int:
```
form.quota_bytes.default = int(app.config['DEFAULT_QUOTA'])
```

## Replication Steps
Set a the environment variable DEFAULT_QUOTA to some value and set a maximum quota on a domain. Now try to create a mail account and the error happens.

## Observed behaviour
When creating a user with DEFAULT_QUOTA set in the environment and a maximum quota set for a domain, the following error is thrown when trying to create a user in that domain:

```
ERROR in app: Exception on /admin/user/create/gunet.de [GET]
Traceback (most recent call last):
  File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/flask_restx/api.py", line 672, in error_router
    return original_handler(e)
           ^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
         ^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/venv/lib/python3.12/site-packages/flask_login/utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/mailu/ui/access.py", line 28, in wrapper
    return callback(function, args, kwargs, dargs, dkwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/mailu/ui/access.py", line 19, in callback
    return function(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/app/mailu/ui/views/users.py", line 31, in user_create
    if form.quota_bytes.default > domain.max_quota_bytes:
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: '>' not supported between instances of 'str' and 'int'
```
## Expected behaviour
No error when creating a mail account

## Logs
See above

### LINKED ISSUE #3401: "Request failed" in Snappymail when displaying one specific INBOX folder
## Environment & Version

- `docker compose version` : `v2.29.1`
- Version: `2024.06.12`

## Description

For one specific INBOX folder or one specific email account, Snappymail fails to displays the INBOX folder contents, and displays "_Request failed_" instead.

Displaying content of other folders of the same account work, and other accounts I tested have no such problem either.

Using an external IMAP client to browse this INBOX folder also works perfectly fine.

## Replication Steps

This started happening for no clear reason − although I did have no space remaining on disk shortly before this problem happened (and since then I fixed the space issue).

## Observed behaviour

When I open the webmail with the faulty account, I get this:

image

In Firefox console logs, I see there is a `500` error, although there is a response which, if I expand it, does contain the list of emails that the UI should display:

image

## Expected behaviour

I expect the INBOX folder to be displayed as before.

## Logs

In the webmail logs I see this message each time I try to refresh the webmail in the faulty INBOX:

```
2024/09/04 08:43:27 [error] 17#17: *366 FastCGI sent in stderr: "PHP message: PHP Fatal error:  [snuffleupagus][0.0.0.0][disabled_function][drop] Aborted execution on call of the function 'ini_set', because its argument '$option' content (display_errors) matched a rule in /var/www/snappymail/snappymail/v/2.36.4/app/libraries/snappymail/shutdown.php on line 15" while reading response header from upstream, client: 37.65.66.89, server: , request: "POST /?/Json/&q[]=/0/ HTTP/1.1", upstream: "fastcgi://unix:/var/run/php8-fpm.sock:", host: "mail.enevned.fr"
```

### LINKED ISSUE #3403: LMTP delivery failing with 'KeyError' in admin logs after upgrading to 2024.06
## Environment & Version

- `docker compose version v2.29.2`
- Version: `2024.06`
- OS: Ubuntu 22.04.4 LTS

## Description

Local delivery fails with an LMTP error

## Replication Steps

I updated to 2024.06 using the upgrade porcedure from 2.0.
Edit: maybe relevant: `INBOUND_TLS_ENFORCE` is set to `True`

Original container versions:
```
ghcr.io/mailu/nginx       2.0            8e086f3810ff   3 months ago    261MB
ghcr.io/mailu/admin       2.0            fb0e93452062   3 months ago    240MB
ghcr.io/mailu/clamav      2.0            32d93905efbc   3 months ago    287MB
ghcr.io/mailu/oletools    2.0            a7e500342676   3 months ago    237MB
ghcr.io/mailu/webmail     2.0            9ac41e28fc71   3 months ago    515MB
ghcr.io/mailu/radicale    2.0            799bbf72b56a   3 months ago    228MB
ghcr.io/mailu/rspamd      2.0            f5143134e95d   3 months ago    289MB
ghcr.io/mailu/unbound     2.0            1023aae15539   3 months ago    238MB
ghcr.io/mailu/dovecot     2.0            f5a96ccb2a1f   3 months ago    307MB
ghcr.io/mailu/postfix     2.0            6a91d0bef0d1   3 months ago    239MB
ghcr.io/mailu/fetchmail   2.0            b9ebf94fb197   3 months ago    234MB
```

## Observed behaviour

Incoming email works, but when postfix tries to deliver to mailbox, the following sequence of events occurs:

1. Front does a passdb lookup
2. Admin logs an error on the /internal/auth/email endpoint
3. Smtp delays delivery due to a temporary lookup error (the passdb lookup that fails)
## Expected behaviour

Delivery works.

## Logs

```
front-1            | Sep 05 09:20:45 lmtp(44912): Info: Connect from 192.168.203.5
front-1            | Sep 05 09:20:45 lmtp(44912): Debug: auth-master: passdb lookup(user@domain.nl): Started passdb lookup
front-1            | Sep 05 09:20:45 lmtp(44912): Debug: auth-master: conn unix:/run/dovecot/auth-userdb: Connecting
front-1            | Sep 05 09:20:45 lmtp(44912): Debug: auth-master: conn unix:/run/dovecot/auth-userdb (pid=12,uid=0): Client connected (fd=16)
front-1            | Sep 05 09:20:45 auth-worker(44910): Error: conn unix:auth-worker (pid=44891,uid=90): auth-worker<2>: lua(user@domain.nl,192.168.203.5): db-lua: 
front-1            | Sep 05 09:20:45 lmtp(44912): Error: auth-master: passdb lookup(user@domain.nl): Auth PASS lookup failed
front-1            | Sep 05 09:20:45 lmtp(44912): Debug: auth-master: passdb lookup(user@domain.nl): auth PASS input: 
front-1            | Sep 05 09:20:45 lmtp(44912): Debug: auth-master: passdb lookup(user@domain.nl): Passdb lookup failed
front-1            | Sep 05 09:20:45 lmtp(44912): Info: Disconnect from 192.168.203.5: Logged out (state=READY)
smtp-1             | Sep 05 09:20:45 mbox02 postfix/qmgr[369]: 4CBD81000C1: from=<SRS0=PQwK=QD=gmail.com=testaddress@mbox02.xcons.nl>, size=4464, nrcpt=1 (queue active)
smtp-1             | INFO:root:Connect 
smtp-1             | Sep 05 09:20:45 mbox02 postfix/lmtp[36688]: 56CBB100039: to=<user@domain.nl>, relay=front[192.168.203.2]:2525, delay=162585, delays=162585/0.06/0.09/0.02, dsn=4.3.0, status=deferred (host front[192.168.203.2] said: 451 4.3.0 <user@domain.nl> Temporary user lookup failure (in reply to RCPT TO command))
admin-1            | [2024-09-05 09:20:45,420] ERROR in app: Exception on /internal/auth/email [GET]
admin-1            | Traceback (most recent call last):
admin-1            |   File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 1473, in wsgi_app
admin-1            |     response = self.full_dispatch_request()
admin-1            |                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
admin-1            |   File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 882, in full_dispatch_request
admin-1            |     rv = self.handle_user_exception(e)
admin-1            |          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
admin-1            |   File "/app/venv/lib/python3.12/site-packages/flask_restx/api.py", line 672, in error_router
admin-1            |     return original_handler(e)
admin-1            |            ^^^^^^^^^^^^^^^^^^^
admin-1            |   File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 880, in full_dispatch_request
admin-1            |     rv = self.dispatch_request()
admin-1            |          ^^^^^^^^^^^^^^^^^^^^^^^
admin-1            |   File "/app/venv/lib/python3.12/site-packages/flask/app.py", line 865, in dispatch_request
admin-1            |     return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
admin-1            |            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
admin-1            |   File "/app/mailu/internal/views/auth.py", line 32, in nginx_authentication
admin-1            |     headers = nginx.handle_authentication(flask.request.headers)
admin-1            |               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
admin-1            |   File "/app/mailu/internal/nginx.py", line 79, in handle_authentication
admin-1            |     status, code = get_status(protocol, "encryption")
admin-1            |                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
admin-1            |   File "/app/mailu/internal/nginx.py", line 137, in get_status
admin-1            |     return status, codes[protocol]
admin-1            |                    ~~~~~^^^^^^^^^^
admin-1            | KeyError: 'lmtp'
```

### LINKED ISSUE #3405: Managing of external certificates - frontend reload
In docs there is an example script for managing external (LE) certificates:

```
#!/bin/sh
cp /etc/letsencrypt/live/domain.com/privkey.pem /mailu/certs/key.pem || exit 1
cp /etc/letsencrypt/live/domain.com/fullchain.pem /mailu/certs/cert.pem || exit 1
docker exec mailu_front_1 nginx -s reload
```

https://mailu.io/master/maintain.html#external-certs

But there is not only nginx but also dovecot in the mailu_front container. So after certificate renew you have to reload dovecot:

`docker exec mailu_front_1 dovecot reload`

Without that you have still old certificate for dovecot services.

### PR DESCRIPTION
**Pull request by nextgens**
## What type of PR?

bug-fix

## What does this PR do?

Update the documentation: clarify that dovecot also needs to be reloaded if custom certs are in use
Fix a 'fatal error' that may be encountered when using snappymail
Fix ``INBOUND_TLS_ENFORCE`` (something you should never use)
Fix ``DEFAULT_QUOTA``
Increase the size of php/nginx buffers on webmail
Maybe fix utf-8 decoding problems in socrate
Maybe fix utf-8 problems in fetchmail folder names

### Related issue(s)
- closes #3401 
- closes #3405
- closes #3403 
- closes #3379
- closes #3272
- closes #2996
- #3398

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [x] In case of feature or enhancement: documentation updated accordingly
- [x] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

**Comment by nextgens** [comment]
bors try

Any objection to me getting rid of ``INBOUND_TLS_ENFORCE`` in master?

**Comment by WebSpider** [comment]
Maybe @lub has an opinion on this?

**Comment by nextgens** [comment]
bors try

**Review by ghostwheel42 — DISMISSED** [review]
Hi. Thanks, LGTM except for two small issues.

File: optional/fetchmail/fetchmail.py [hunk 1]
```diff
@@ -32,6 +33,21 @@
...
+    for c in s:
+        if 0x20<=ord(c)<=0x7f :
+            if unipart!='' :
+                out+='&'+base64.b64encode(unipart.encode('utf-16-be')).decode('ascii').rstrip('=')+'-'
+                unipart=''
+            out+=c
+        else : unipart+=c
+    if unipart!='' :
+        out+='&'+base64.b64encode(unipart.encode('utf-16-be')).decode('ascii').rstrip('=')+'-'
+    return out
```
[the remark this answers is not in the document]
Comment (nextgens) [reply optional/fetchmail/fetchmail.py:50]: ``SyntaxError: f-string expression part cannot include a backslash``

Don't hesitate to push more commits to the branch directly

[the remark this answers is not in the document]
Comment (ghostwheel42) [reply optional/fetchmail/fetchmail.py:50]: Hmm. Python < 3.12. Too bad.

[the remark this answers is not in the document]
Comment (nextgens) [reply optional/fetchmail/fetchmail.py:50]: Hmm, that was my host system, alpine has python 3.12

**Comment by nextgens** [comment]
bors try

File: optional/fetchmail/fetchmail.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (ghostwheel42) [reply optional/fetchmail/fetchmail.py]: Okay, I've backported it to python 3.10 now.

**Pull request merged**